atc/configvalidate/validate_test.go

```diff
@@ -661,214 +730,95 @@ var _ = Describe("ValidateConfig", func() {
+				Expect(errorMessages[0]).To(ContainSubstring("jobs.some-other-job.plan.do[1].get(some-resource): repeated name"))
 			})
 		})
 
-		Context("when a job has duplicate inputs via parallel", func() {
+		Context("when a job gets the same resource multiple times but with different names", func() {
 			BeforeEach(func() {
-				job.PlanSequence = append(job.PlanSequence, PlanConfig{
-					Get: "some-resource",
+				job.PlanSequence = append(job.PlanSequence, atc.Step{
+					Config: &atc.GetStep{
+						Name:     "a",
+						Resource: "some-resource",
+					},
 				})
-				job.PlanSequence = append(job.PlanSequence, PlanConfig{
-					InParallel: &InParallelConfig{
-						Steps: PlanSequence{
-							{
-								Get: "some-resource",
-							},
-						},
-						Limit:    1,
-						FailFast: true,
+				job.PlanSequence = append(job.PlanSequence, atc.Step{
+					Config: &atc.GetStep{
+						Name:     "b",
+						Resource: "some-resource",
 					},
 				})
 
 				config.Jobs = append(config.Jobs, job)
 			})
 
-			It("returns a single error", func() {
-				Expect(errorMessages).To(HaveLen(1))
-				Expect(errorMessages[0]).To(ContainSubstring("invalid jobs:"))
-				Expect(strings.Count(errorMessages[0], "has get steps with the same name: some-resource")).To(Equal(1))
+			It("returns no errors", func() {
+				Expect(errorMessages).To(HaveLen(0))
 			})
 		})
 
 		Describe("plans", func() {
-			Context("when multiple actions are specified in the same plan", func() {
-				Context("when it's not just Get and Put", func() {
-					BeforeEach(func() {
-						job.PlanSequence = append(job.PlanSequence, PlanConfig{
-							Get:        "some-resource",
-							Put:        "some-resource",
-							Task:       "some-resource",
-							Do:         &PlanSequence{},
-							Aggregate:  &PlanSequence{},
-							InParallel: &InParallelConfig{},
-						})
-
-						config.Jobs = append(config.Jobs, job)
-					})
-
-					It("returns an error", func() {
-						Expect(errorMessages).To(HaveLen(1))
-						Expect(errorMessages[0]).To(ContainSubstring("invalid jobs:"))
-						Expect(errorMessages[0]).To(ContainSubstring("jobs.some-other-job.plan[0] has multiple actions specified (aggregate, do, get, parallel, put, task)"))
-					})
-				})
-
-				Context("when it's just Get and Put (this was valid at one point)", func() {
-					BeforeEach(func() {
-						job.PlanSequence = append(job.PlanSequence, PlanConfig{
-							Get:        "some-resource",
-							Put:        "some-resource",
-							Task:       "",
-							Do:         nil,
-							Aggregate:  nil,
-							InParallel: nil,
-						})
-
-						config.Jobs = append(config.Jobs, job)
-					})
-
-					It("returns an error", func() {
-						Expect(errorMessages).To(HaveLen(1))
-						Expect(errorMessages[0]).To(ContainSubstring("invalid jobs:"))
-						Expect(errorMessages[0]).To(ContainSubstring("jobs.some-other-job.plan[0] has multiple actions specified (get, put)"))
```

Comment:
Good point. I pushed a change that puts the step type to the message, at least, so it now reads like so:

```
unmarshal converted json: malformed get step: json: unknown field "bogus"
```

Still not great, but it doesn't look like there's an easy way to show anything more. :/ We won't have an accurate line number to report since the YAML is first converted to JSON. It'd be nice if `encoding/json` returned more semantic information in these cases.